atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraServersConfigs.java

```diff
@@ -42,18 +44,16 @@ private CassandraServersConfigs() {
     private static final String SERVER_FORMAT_ERROR = "each server must specify a port ([host]:[port])";
     private static final String PORT_NUMBER_ERROR = "%s port number should be a positive number";
 
-    public interface Visitor<T> extends BiFunction<Set<InetSocketAddress>, Optional<Set<InetSocketAddress>>, T> {
-        @Override
-        T apply(Set<InetSocketAddress> thriftServers, Optional<Set<InetSocketAddress>> maybeCqlServers);
-    }
 
     @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type", defaultImpl = ImmutableDefaultConfig.class)
     @JsonSubTypes({
             @JsonSubTypes.Type(value = ImmutableDefaultConfig.class, name = DefaultConfig.TYPE),
-            @JsonSubTypes.Type(value = ImmutableCqlCapableConfig.class, name = CqlCapableConfig.TYPE)
-            })
+            @JsonSubTypes.Type(value = ImmutableCqlCapableConfig.class, name = CqlCapableConfig.TYPE)})
     public interface CassandraServersConfig {
-        <T> T visit(Visitor<T> visitor);
+
+        <T> T visitThrift(Function<Set<InetSocketAddress>, T> visitor);
+
+        <T> T visitCql(BiFunction<Optional<Set<InetSocketAddress>>, Optional<SocketAddress>, T> visitor);
```

Comment:
Have a proper visitor, (as a new reader) I have no idea what this is supposed to mean, they're all `SocketAddress`'s!

If you insist on no visitors, at the very least have a method that exposes:
* thrift servers
* some optional object encapsulating the cql ports and the optional proxy
  * note this is just the object itself, at this point you should probably have a visitor

Reply:
Implemented a visitor.